datafusion-cli/src/print_format.rs

```diff
@@ -159,14 +191,41 @@ impl PrintFormat {
         }
         Ok(())
     }
+
+    pub fn print_stream(&self, batch: &RecordBatch, with_header: bool) -> Result<()> {
```

Comment:
This function is almost a copy/paste of `print_batch` -- as you say in your comments, the only difference is i Self::Table.  I think it would be possible to reduce replication by changing all this code to be in terms of the output stream rather than creating strings

Reply:
Specifically, I think you could write the functions in terms of `std::io::Write` and then pass in https://doc.rust-lang.org/std/io/fn.stdout.html to avoid having to both copy the code and the code could avoid copying strings